assets/src/components/reader-theme-carousel/index.js

```diff
@@ -0,0 +1,170 @@
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import { AMP_QUERY_VAR, DEFAULT_AMP_QUERY_VAR, LEGACY_THEME_SLUG, AMP_QUERY_VAR_CUSTOMIZED_LATE } from 'amp-settings'; // From WP inline script.
+
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { useContext, useMemo, useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { CheckboxControl } from '@wordpress/components';
+import { ReaderThemes } from '../reader-themes-context-provider';
+import { Loading } from '../loading';
+import './style.css';
+import { AMPNotice, NOTICE_TYPE_WARNING } from '../amp-notice';
+import { ThemeCard } from '../theme-card';
+import { Carousel } from '../carousel';
+
+/**
+ * Component for selecting a reader theme.
+ *
+ * @param {Object} props Component props.
+ * @param {boolean} props.hideCurrentlyActiveTheme Whether the currently active theme should be unselectable.
+ */
+export function ReaderThemeCarousel( { hideCurrentlyActiveTheme = false } ) {
+	const { currentTheme, fetchingThemes, selectedTheme, themes: unprocessedThemes } = useContext( ReaderThemes );
+
+	const [ includeUnavailableThemes, setIncludeUnavailableThemes ] = useState( false );
+
+	const { activeTheme, themes } = useMemo( () => {
+		let active, processedThemes;
+
+		if ( hideCurrentlyActiveTheme ) {
+			processedThemes = ( unprocessedThemes || [] ).filter( ( theme ) => {
+				if ( 'active' === theme.availability ) {
+					active = theme;
+					return false;
+				}
+				return true;
+			} );
+		} else {
+			active = null;
+			processedThemes = unprocessedThemes;
+		}
+
+		return { activeTheme: active, themes: processedThemes };
+	}, [ hideCurrentlyActiveTheme, unprocessedThemes ] );
```

Comment:
I see this was copied from [assets/src/components/reader-theme-selection/index.js](https://github.com/ampproject/amp-wp/blob/f5b4b3586c9660c2276c24f64abbd688be9c8a4a/assets/src/components/reader-theme-selection/index.js#L31-L48). I think it would be better to extract the logic into an exported function and import it here (to keep it DRY).

Reply:
Done in 92b28f58f